### DISCUSSION TITLE
Unexpected chromaticities plot while using Chromaticity Diagram.

### DISCUSSION THREAD

**Discussion by matthiasbirkich** [colour-science/colour#732] [Q&A] [answered]
Dear developers, 

thank you very much for developing the colorscience library for python.
I have tested a couple of calculators and imaging tools for color science to proof the python library.
In my opinion in one point the software is not doing the right.
The plot of xy-chromaticity values in the xy-diagram will not reflect the right coordinates, when it is used with the command:

plot_RGB_chromaticities_in_chromaticity_diagram_CIE1931

Are you so polite to tell me, which commands I have to use to plot xy-chromaticity values into the chromaticity diagram?
In the pdf is the code for two different codings and the graphs.
The dot is representing the Macbeth color checker No14 green.
Thank you very much for your work and your answer.

best regards
Matthias
![18D1048A-1F82-4B46-B032-A1971710A3D5](https://user-images.githubusercontent.com/59750408/72192270-598b3e80-3405-11ea-8c91-5679caca8200.png)
![4B317DFB-67B9-46B3-B60F-B18ED1C406BE](https://user-images.githubusercontent.com/59750408/72192271-5a23d500-3405-11ea-94bf-0bfbcb35c5e2.png)
[01_ColorCheckerGreen.pdf](https://github.com/colour-science/colour/files/4048498/01_ColorCheckerGreen.pdf)

**Comment by thomasmansencal** [comment]
Hi @matthiasbirkich,

I'm not sure to understand this part:

> The plot of xy-chromaticity values in the xy-diagram will not reflect the right coordinates, when it is used with the command:

Would you mind clarifying please?

**Comment by matthiasbirkich** [comment]
Dear KelSolar, thank you for your fast response.
The xy-chromaticity-coordinates are 0.30654595, 0.48251036.
In the diagram the coordinates are laying at 0.30, 0.425. This circumstance I does not understand.
Best regards
Matthias

**Comment by nick-shaw** [comment, marked as answer]
`plot_RGB_chromaticities_in_chromaticity_diagram_CIE1931` expects linear RGB values in the specified colour space, and converts them to CIExy for plotting. What is your input?

**Comment by matthiasbirkich** [comment]
Dear nick-shaw,
ok, that’s what I expected, but not proofed. Sorry!
I thought RGB is something like 0-255. But here it is %RGB/100...
My input code is: 
importcolour
import matplotlib.pyplot as plt
import numpy as np
from colour.plotting import *
from colour.utilities import message_box
from pprint import pprint
xyY = np.array([0.30654595, 0.48251036, 0.2328437852])
xy = print(colour.xyY_to_xy(xyY))
RGB = xyY
plot_RGB_chromaticities_in_chromaticity_diagram_CIE1931(RGB,'S-Gamut', colourspaces=['S-Gamut'],whitepoint = False, show_diagram_colours=False)
plt.clf() # clear memory of graphs
——
When I convert this to xyY values, I get the values which are in the diagram: xy = 0.323895, 0.405664. This is consistent.

Does there is a command, which is plotting xy-chromaticity values directly into the chromaticity diagram with colorspace triangle? I want to choose the dotsize, dotcolor and the diemsions of the x-axis and y-axis as well. Thank you for your help!

best regards, Matthias

**Comment by matthiasbirkich** [comment]
What I also not understand:
RGB is 3dimensional, xy-chromaticity plots are definitely 2dimensional. The RGB output from the xyY input is: 
0.323895, 0.405664,  0.000130. Does Y is always negligible here?
best regards Matthias

**Comment by matthiasbirkich** [comment]
Thanks for your support, I will go on learning Python language ;-)

**Comment by thomasmansencal** [comment]
Hey @matthiasbirkich,

The Chromaticity Diagram(s) is(are) top-down view(s) along the Luminance/Lightness axis of a particular colour model/space. The CIE 1931 Chromaticity Diagram uses CIE xy**Y** colour model/space, CIE 1976 UCS uses CIE **L**uv. I see that you closed the issue, was your issue resolved then?

Cheers,

Thomas

**Comment by matthiasbirkich** [comment]
Dear Troy,

thank you for your answer.
I haven’t recognized your comment.
Now the issue is solved.
Stay healthy and happy!
best regards
Matthias

Troy James Sobotka <notifications@github.com> schrieb am Sa. 11. Jan. 2020
um 02:06:

> Should be rather simple. The xy chromaticity diagram is given via several
> plotting functions. For your case, I think you are looking for the three
> primaries as expressed in xy coordinates.
>
> You can use the built in primaries, or optionally, make your own array. It
> should be something like this for example, using the perceptually uniform
> 1976 plot:
>
> colour.plotting.plot_RGB_colourspaces_in_chromaticity_diagram_CIE1976UCS('ITU-R BT.709', diagram_clipping_path=colour.xy_to_Luv_uv(colour.models.RGB_COLOURSPACES['ITU-R BT.709'].primaries), bounding_box=(-.1, .7, -.05, .7), spectral_locus_colours="RGB")
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/colour-science/colour/issues/545?email_source=notifications&email_token=AOH3QCAZB3RDJLZJ3FQ2KALQ5ELRVA5CNFSM4KFOLQX2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEIVUWIQ#issuecomment-573262626>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AOH3QCAXFHESOBCIQTGN6S3Q5ELRVANCNFSM4KFOLQXQ>
> .
>
-- 
--
Matthias Birkicht
Georg-Seebeck-Straße 47
27570 Bremerhaven
mobile: 01609 48 68 28 7
phone: +49 471 306608
work: +49 421 23 800 34
skype: matthias.birkicht
www.oberton.info
